opensrp-giz-malawi/src/main/java/org/smartregister/giz/view/NavigationMenu.java

```diff
@@ -0,0 +1,333 @@
+                    ((ViewGroup) current.getParent()).removeView(current); // <- fix
+                }
+
+                // swap content view
+                LayoutInflater mInflater = LayoutInflater.from(activity);
+                ViewGroup contentView = (ViewGroup) mInflater.inflate(R.layout.navigation_drawer, null);
+                activity.setContentView(contentView);
+
+                rootView = activity.findViewById(R.id.nav_view);
+                RelativeLayout mainView = activity.findViewById(R.id.nav_content);
+
+                if (current.getParent() != null) {
+                    ((ViewGroup) current.getParent()).removeView(current); // <- fix
+                }
+
+                if (current instanceof RelativeLayout) {
+                    RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(
+                            RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT);
+                    params.addRule(RelativeLayout.ALIGN_PARENT_LEFT, RelativeLayout.TRUE);
+                    current.setLayoutParams(params);
+                    mainView.addView(current);
+                } else {
+                    mainView.addView(current);
+                }
+            } else {
+                rootView = current;
+            }
+        }
+    }
+
+    private void registerDrawer(Activity parentActivity) {
+        if (drawer != null) {
+            drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED);
+            ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+                    parentActivity, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+            drawer.addDrawerListener(toggle);
+            toggle.syncState();
+
+        }
+    }
+
+    @Override
+    public void prepareViews(Activity activity) {
+        drawer = activity.findViewById(R.id.drawer_layout);
+        recyclerView = rootView.findViewById(R.id.rvOptions);
+        // NavigationView navigationView = rootView.findViewById(R.id.nav_view);
+        tvLogout = rootView.findViewById(R.id.tvLogout);
+        recyclerView = rootView.findViewById(R.id.rvOptions);
+        ivSync = rootView.findViewById(R.id.ivSyncIcon);
+        syncProgressBar = rootView.findViewById(R.id.pbSync);
+        ImageView ivLogo = rootView.findViewById(R.id.ivLogo);
+        ivLogo.setContentDescription(activity.getString(R.string.nav_logo));
+        ivLogo.setImageResource(R.drawable.ic_logo);
+
+        TextView tvLogo = rootView.findViewById(R.id.tvLogo);
+        tvLogo.setText(activity.getString(R.string.nav_logo));
+
+        if (syncProgressBar != null) {
+            FadingCircle circle = new FadingCircle();
+            syncProgressBar.setIndeterminateDrawable(circle);
+
+            if (toolbar != null) {
+                toolbar.setNavigationIcon(circle);
+            }
+        }
+
+        // register all objects
+        registerDrawer(activity);
```

Comment:
This is being called again on `Line 90` in the same file before this method `prepareViews` is called. I am guessing it should be called only once

Reply:
Fixed!